modules/src/applications/ics20_fungible_token_transfer/msgs/transfer.rs

```diff
@@ -23,9 +25,9 @@ pub struct MsgTransfer {
     /// the channel by which the packet will be sent
     pub source_channel: ChannelId,
     /// the tokens to be transferred
-    pub token: Option<ibc_proto::cosmos::base::v1beta1::Coin>,
+    pub token: IbcCoin,
     /// the sender address
-    pub sender: Signer,
+    pub sender: Address,
     /// the recipient address on the destination chain
     pub receiver: Signer,
```

Comment:
Why this receiver Type is not Address?

Reply:
The `Address` is a valid `bech32` encoded address. I think the handlers should treat it as opaque data and leave such validation up to the host chain, so technically both `sender` and `receiver` should be `Signer` types and I am currently changing them both to be `Signer`. But we need to communicate that the address needs validation through the API and document it. I think the best place to document this would be the `FromStr` impl for `AccountId` ->
https://github.com/informalsystems/ibc-rs/pull/1989/files#diff-8f9bde3d68bd2354e1fa18e322a80e0c6d0b595d6b281db282116f948b44b8d5R137